### DISCUSSION TITLE
VSCode - Missing DLLs when using VSCode-related tools to install

### DISCUSSION THREAD

**Discussion by Boejunda** [cefsharp/CefSharp#4223] [General]
- **What version of the product are you using?**

104.4.180

- **What architecture x86 or x64?**

    All architectures tested: x86, x84 and "ANYCPU"
	
- **What version of .Net?**

- .Net 4.0 
- .Net 4.8 
- .Net 6.0

    
- **On what operating system?**

- Win10

- **Are you using `WinForms`, `WPF` or `OffScreen`?**

WinForms

    
- **What steps will reproduce the problem?**
- When using **VSCode**-related tools to install CefSharp.Winforms, certain DLLs do not install.
- Tried using the command:  "dotnet add package CefSharp.WinForms --version 104.4.180".
- Also tried installing the NuGet Package Manager extension. Behavior is that CefSharp can be found, and it's variants ("CefSharp.Winforms" for example), but that tool seemingly cannot read the version information and fails to download at all.

/* Tried using a minimal example . Code: */

using System.Windows.Forms;
using CefSharp.WinForms;

public class FooForm : Form {
     private ChromiumWebBrowser browser;

     public FooForm () {
          // Initialize the browser, here or after the constructor....
          browser = new ChromiumWebBrowser("www.foobar.com");
          // Above line fails with a "Can't find CefSharp.dll" - style error message.
     }
}

- **What is the expected output? What do you see instead?**

I expected the ChromiumWebBrowser object to instantiate and my interface to load with a web browser interface to work with.
Instead, I see a runtime error that indicates a central library, CefSharp.dll cannot be found.

- **Please provide any additional information below.**
    - A stack trace if available, any Exception information.
      - If you are seeing a crash in `libcef.dll` then please download `libcef.dll.pdb` and place it next to `libcef.dll` to obtain a detailed stack trace, see https://github.com/cefsharp/CefSharp/wiki/Trouble-Shooting#loading-native-symbols-for-easier-diagnosis for details.

Exception here:

An exception of type 'System.IO.FileNotFoundException' occurred in Story Writing.dll but was not handled in user code: 'Could not load file or assembly 'CefSharp, Version=104.4.180.0, Culture=neutral, PublicKeyToken=40c4b6fc221f4138'. The system cannot find the file specified.'

- Does the cef log provide any relevant information? (By default there should be a debug.log file in your bin directory)

I never was able to get CefSharp running in a VSCode project, so those logs were never generated.

- Any other background information that's relevant? Are you doing something out of the ordinary? 3rd party controls?

Again, I was using VSCode to build my original project before I tried to implement this. Having had a great deal of success for a variety of projects, I stopped using VS some time ago. Since encountering this issue, it seems that VS has some built-in tools that VSCode does not have sufficient corollaries for. All the same, I think it's worth examining why that is, and whether or not there is additional guidance by the CefSharp team in this area. I hope the information I've provided helps.

**Comment by amaitland** [comment]
>  Since encountering this issue, it seems that VS has some built-in tools that VSCode does not have sufficient corollaries

Converting to discussion as this is unlikely something that can be fixed in `CefSharp`. If it turns out code changes are require a new issue can be created that clearly details those code changes.

If further documentation is required this discussion can document the process.

>What version of .Net?

.Net 4.0 - Not supported, minimum is `.Net 4.5.2`

.Net 6.0 - You need to use [CefSharp.WinForms.NETCore](https://www.nuget.org/packages/CefSharp.WinForms.NETCore) when targeting `.Net 5.0 or greater`

> When using VSCode-related tools 

What are `VSCode-related tools` exactly? I've never used `VSCode` for `C#` dev. If you are referring to `dotnet.exe` that's part of the `.Net SDK` and not specific to `VSCode`.

> I hope the information I've provided helps.

Please provide a clear step by step list of exactly the tasks/commands you've performed. As I said, never used `VSCode` for `.Net` dev, so not sure what you have actually done/tried.
